src/DynamoCoreWpf/Views/Menu/PreferencesView.xaml.cs

```diff
@@ -37,9 +38,59 @@ public PreferencesView(DynamoViewModel dynamoViewModel)
                 viewModel = viewModelTemp;
             }
 
+            SetAllExpanderBindings();
+
             InitRadioButtonsDescription();
         }
 
+        /// <summary>
+        /// Due that all the settings for the expanders is stored in DynamoModel in a List property, we need to create expanders bindings dynamically
```

Comment:
I think you can do a few things as alternatives if you want.
* use dataTemplates and don't manually create each expander, instead store data that represents each one, including the binding.
* use a valueConverter that converts the list of expanders to a specific expander depending on the name of the expander - this will let you add the binding into the xaml if you prefer.

just some other ideas.

Reply:
@mjkkirschner 
Related to the option of using dataTemplates, It will take some effort since we have already 6 expanders in 2 different tabs so the way expanders are created (considering each expander has several controls) will change and probably how the data is bind.
Related to the option of using a converter, I was trying to implement the converter but currently is using two way binding, so at the beginning the values are read from the List<ExpanderSettings> and when a expander is expanded the IsExpanded value is stored in the List<ExpanderSettings>, then when implementing the ConverterBack function I didn't find any way to store the IsExpanded value updated, since it doesn't have access to the List.

The considering the function below, the value parameter is the boolean IsExpanded value and the parameter is the Expander Name, so there is no way to access the List<ExpanderSettings> as in the Convert( ) method.

**ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)**

If you have any idea how to make it work and I can implement it.

Thanks

Reply:
It is unclear to me why the expanders stored in DynamoModel. It makes most sense to me to store them in Preferences ViewModel so that in the future those code maintained as part of PreferencesView. Can we revisit the base of this PR?

Reply:
@QilongTang In this case that we are trying to record expansion state of each expander per session so I cannot use the PreferencesViewModel because is created just when the Preferences window is opened and destroyed when is closed then the only way to restore/record the expansion state of each expander is by storing the instance in a class that is not destroyed, in this case I decided to use DynamoModel (like DebugSettings or PreferencesSettings instances), I can use other one but it's not clear to me the idea of using PreferencesViewModel.

Reply:
hi @RobertGlobant20 , let's do the right thing. We can create preference viewmodel early as part of Dynamo ViewModel creation instead of only construct/destroy it runtime. Then we can store these properties there right?

Reply:
@QilongTang I did the required changes so now the PreferencesViewModel is created in DynamoViewModel.cs and the properties for the expanders were moved to the PreferencesViewModel. Thanks
See the commit:
https://github.com/DynamoDS/Dynamo/pull/11689/commits/353a4988e1b4bc4cd5110521be9688e79c88a47e